tests/testing.ts

```diff
@@ -252,8 +257,11 @@ const expectedResults = new Map<string, ResultType>(
 export function runTests(force: boolean, onlyResults: boolean, unsafe = false, exact = false) {
   const results = new Map<string, ResultType>();
   if (force) {
+    if (!cairo1setup()) doCairo1Setup();
     postTestCleanup();
-  } else if (!preTestChecks()) return;
+  } else {
+    if (!cairo1setup() || !preTestChecks()) return;
+  }
```

Comment:
Let's check for the presence of binaries first

Reply:
cairo1 does check for presence of binaries, line -307

Reply:
Oh I see. As I think about that. Why don't we upload binaries to repo until they're available via package manager? It's no use to compile them over and over again. Even for mac ARM users one person can compile it so others won't have to

Reply:
but we need to upload binaries for all available different architectures and machines, which is pretty tough task , also we have to detect the system architecture during runtime and download required binaries related to specific architecture.

Reply:
I think we already do that for nethersolc and other test utilities from solidity, what is the difference? Imo it's maximum 4 different binaries (arm linux, x86 linux, arm mac, x86 mac). Anyway that's not a big deal. I assume in the long term it will be distributed via package manager so we will manage without binaries in repo